app/bot/__init__.py

```diff
@@ -110,58 +110,50 @@ async def on_ready():
 async def on_message(message: discord.Message):
     # Don't respond to messages from different guilds if it is enforced
     if not _in_guild(message):
-        if _botconf.botconfig.enforce_guild:
+        if _botconf.bot_config.enforce_guild:
             return
 
     # Don't respond to this bot's own messages
     if message.author == client.user:
         return
 
-    if _is_greeting(message):
-        logger.info("received greeting")
-        greeting = random.choice(_botconf.botconfig.greetings)
-        logger.info(f"greeting: {greeting}")
-        response = await llm.generate_response(
-            message,
-            _botconf.botconfig.system_prompt +
-            " You will respond with a somewhat short greeting and mention their name." +
-            f" You will incorperate the phrase \"{greeting}\" into your greeting." +
-            f" Their name is {message.author.name}.",
-        )
-        logger.info(f"response: `{response}`")
-        if response is not None:
-            await message.reply(response)
-            await client.change_presence(status=discord.Status.online)
-        else:
-            await client.change_presence(status=discord.Status.idle)
-        return
-
     if _is_command(message.content):
         logger.info("received command")
         split_message = message.content.strip(" \t\n").split()
-        command = split_message[0][len(_botconf.botconfig.command_prefix):]
+        command = split_message[0][len(_botconf.bot_config.command_prefix):]
         args = split_message[1:]
 
         await _handle_command(command, args, message)
         return
 
+    # Add message to history
+    _bothist.bot_history.add_message(message)
+
     if client.user in message.mentions:
         logger.info("received message")
+
         response = await llm.generate_response(
             message,
-            _botconf.botconfig.system_prompt +
-            f" The user's name is {message.author.name}",
+            _botconf.bot_config.system_prompt,
```

Comment:
This doesn't make a lot of sense with how chat system prompts work, afaik. Essentially the first message in the chat history will be your system prompt, with a role of `system`, and the content of your system prompt.

Reply:
Yes, I add the system prompt to the messages before generating the response:
https://github.com/btc-raspberrypiclub/piclub-bot/pull/20/files#diff-de86459da278c0199e14e2432900816166309e862e9dee0dd929ae3307acfec4R36